### ISSUE TITLE
List View: Block selection (and focus transfer) works inconsistently based on whether a block is selected

### ISSUE THREAD

**Issue by talldan** [WordPress/gutenberg#66276]
### Description

List View serves as an important accessibility tool, users can jump into List View using a shortcut, navigate blocks using arrow keys and usually jump back to a block in the canvas by selecting it in List View.

There's one case where this doesn't quite work - if a block is already selected, if you try selecting it again focus doesn't transfer to the canvas.

The semantics in List View don't really tell you this wouldn't work, all the block items are links to the blocks in the canvas.

The best outcome seems to be making sure this works and focus is transferred.

### Step-by-step reproduction instructions

1. Create a new post and add a few paragraph blocks
2. Press <kbd>Ctrl</kbd>+<kbd>Opt</kbd>+<kbd>O</kbd> (on Mac) or <kbd>Alt</kbd>+<kbd>Shift</kbd>+<kbd>O</kbd> (on Windows/Linux) to open List View and note that the select block is focused in List View
3. Navigate to another block and select it using <kbd>Enter</kbd> or <kbd>Space</kbd>, note that it moves focus back to the canvas
4. Open List View again (<kbd>Ctrl</kbd>+<kbd>Opt</kbd>+<kbd>O</kbd> or <kbd>Alt</kbd>+<kbd>Shift</kbd>+<kbd>O</kbd>)
5. Try selecting the already selected block using <kbd>Enter</kbd> or <kbd>Space</kbd>
6. Observe that focus isn't moved.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

**Comment by draganescu** [comment]
The focus is transfered when selection changes. But, since the selected block does not change when we select the selected block in the list view, there is no selection change. Therefore no focus transfer.

Also for partially synced patterns the list view only sends focus to the editable blocks if the parent pattern is previously selected, this looks like a separate issue.

**Comment by talldan** [comment]
> The focus is transfered when selection changes. But, since the selected block does not change when we select the selected block in the list view, there is no selection change. Therefore no focus transfer.

Thanks for looking into it. I think that's a good summary of the issue. IIRC, the code that handles the focus is quite decoupled from the action that triggers block selection, so it isn't an easy problem to solve.

> Also for partially synced patterns the list view only sends focus to the editable blocks if the parent pattern is previously selected, this looks like a separate issue.

I couldn't reproduce that on latest `trunk`. I think there might have been some experimentation with sending focus to the first contentOnly block when the parent is selected in the canvas, so maybe that leaked into the list view implementation too.
